### PR TITLE
[Feature] /check-message API for manual smishing detection

### PR DESCRIPTION
**Pull request by harshthadeshwar**
**### What This PR Does**
This pull request implements the Manual Text Checker feature for smishing detection. The following backend functionalities have been added:

- Created a new POST endpoint: POST /api/check-message

- Integrated a Scikit-learn-based trained ML model to detect smishing

- Loaded pre-trained vectorizer and model from .pkl files

- Validates user input (ensures it’s a non-empty string)

- Takes message text input, processes it, and returns:

=> the original message

=> the predicted label (e.g., smishing, safe)

🧪 Testing Done
✅ Tested using Thunder Client with real smishing examples

✅ Received expected prediction results (200 OK) from the /check-message route

✅ Backend is successfully connected to MongoDB and ML model executes without crash

### CODE REVIEW

**Comment by Seb-sketch** [comment]
This is request is being made for the main branch, please ensure any changes are only requested for the dev branch.

**Comment by harshthadeshwar** [comment]
Why does it shows that this pull request is closed with unmerged commits?
